src/UI/IUIUserInstanceFactory.cs

```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Identity.UI
+{
+    public interface IUIUserInstanceFactory<TUser> where TUser : class
```

Comment:
This name is pretty meh, maybe we should just call it IUserFactory with a Create method

Reply:
As mentioned above. Let's not introduce an additional abstraction for this and just rely on the parameter-less constructor constraint.